modules/siddhi-extensions/event-table/src/test/java/org/wso2/siddhi/extension/eventtable/hazelcast/DefineTableTestCase.java

```diff
@@ -0,0 +1,471 @@
+
+        HazelcastInstance instance = instanceMap.get(HazelcastEventTableConstants.HAZELCAST_INSTANCE_PREFIX + executionPlanRuntime.getName());
+        Assert.assertEquals(clusterName, instance.getConfig().getGroupConfig().getName());
+        Assert.assertEquals(clusterPassword, instance.getConfig().getGroupConfig().getPassword());
+
+        executionPlanRuntime.shutdown();
+    }
+
+    @Test
+    public void testQuery19() throws InterruptedException {
+        log.info("testTableDefinition19 - OUT 0");
+
+        String clusterName = "siddhi_cluster_t19";
+        String clusterPassword = "cluster_pw";
+
+        Config cfg_1 = new Config();
+        cfg_1.getGroupConfig().setName(clusterName).setPassword(clusterPassword);
+        cfg_1.setProperty("hazelcast.logging.type", "log4j");
+        HazelcastInstance instance_1 = Hazelcast.newHazelcastInstance(cfg_1);
+
+        Config cfg_2 = new Config();
+        cfg_2.getGroupConfig().setName(clusterName).setPassword(clusterPassword);
+        cfg_2.setProperty("hazelcast.logging.type", "log4j");
+        HazelcastInstance instance_2 = Hazelcast.newHazelcastInstance(cfg_2);
+
+        StringBuilder sb = new StringBuilder();
+        for (Member member : instance_2.getCluster().getMembers()) {
+            sb.append(member.getSocketAddress()).append(',');
+        }
+        String addresses = StringUtils.replace(sb.toString(), "/", "");
+        addresses = StringUtils.removeEnd(addresses, ",");
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+        StreamDefinition streamDefinition = StreamDefinition.id("StockStream").attribute("symbol", Attribute.Type.STRING).attribute("price", Attribute.Type.INT);
+        TableDefinition tableDefinition = TableDefinition.id("StockTable")
+                .annotation(Annotation.annotation("from")
+                        .element("eventtable", "hazelcast")
+                        .element("cluster.name", clusterName)
+                        .element("cluster.password", clusterPassword)
+                        .element("cluster.addresses", addresses))
+                .attribute("symbol", Attribute.Type.STRING)
+                .attribute("price", Attribute.Type.INT);
+        Query query = Query.query();
+        query.from(InputStream.stream("StockStream"));
+        query.insertInto("StockTable");
+        ExecutionPlan executionPlan = new ExecutionPlan("ep1");
+        executionPlan.addQuery(query);
+        executionPlan.defineStream(streamDefinition);
+        executionPlan.defineTable(tableDefinition);
+        ExecutionPlanRuntime executionPlanRuntime = siddhiManager.createExecutionPlanRuntime(executionPlan);
+
+        InputHandler stockStream = executionPlanRuntime.getInputHandler("StockStream");
+        executionPlanRuntime.start();
+        stockStream.send(new Object[]{"WSO2", 55.6f});
+        stockStream.send(new Object[]{"IBM", 75.6f});
+        Thread.sleep(RESULT_WAIT * 4);
```

Comment:
@grainier Shall we introduce the 'waitForEvents' logic here as well so that the probability of test cases failing due to slow machines is low?